src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1474,7 +1478,6 @@ RouterQueryJob(Query *query, Task *task, List *placementList)
 	Job *job = NULL;
 	List *taskList = NIL;
 	TaskType taskType = task->taskType;
-	bool requiresMasterEvaluation = RequiresMasterEvaluation(query);
```

Comment:
We can initialize requiresMasterEvalutation to false, and set it if the taskType = MODIFY_TASK few lines below